SpatialGDK/Source/SpatialGDKTests/Examples/SpatialGDKTestGuidelines.h

```diff
@@ -0,0 +1,56 @@
+/*
+0. How to run tests:
+	- Tests can be run via Session Frontend in UE4 Editor
+
+	- Tests can be run via command line
+
+1. Folder structure
+	- Tests are located in a separate SpatialGDKTests module.
+	So for every component that needs to be tested - it has to be exposed via corresponding macro.
+	E.g. `class FRPCContainer` -> `class SPATIALGDK_API FRPCContainer`, to make FRPCContainer testable.
+	WThe macro is different in each module SPATIALGDK_API, SPATIALGDKSERVICES_API etc.
+
+	- The folder structure inside SpatialGDKTests should resemble the folder structure for components being tested.
+	E.g. If file tested is
+	`UnrealGDK\SpatialGDK\Source\SpatialGDK\Public\Utils\RPCContainer.cpp`,
+	then the corresponding test should be located in
+	`UnrealGDK\SpatialGDK\Source\SpatialGDKTests\SpatialGDK\Utils\RPCContainer\RPCContainerTest.cpp`.
+	There should be a folder with the same name as component tested. It should include the test and all the supporting files.
+
+2. Test definitions (check TestDefinitions.h for more info)
+    - We have defined 3 types of Macro to be used when writing tests:
+	(https://docs.unrealengine.com/en-US/Programming/Automation/TechnicalGuide/index.html has more information)
+	TEST - the most common simple test
+	COMPLEX_TEST - same as simple test, but allows having multiple test cases run through the same test function body.
+	DEFINE_LATENT_COMMAND... - used to run tests that require more than 1 thread.
```

Comment:
I thought it was anything that ticks

Reply:
It is. So I think it's mainly applicable to code with multiple threads of executions.

Reply:
You can write something with multiple threads without that in a simple test though.
The main way I've seen it used so far is making sure you get engine related stuff, like a world etc.

Reply:
Unless by thread you mean that task system. Which still feels like it's engine systems interactions but I see what you mean then.

Reply:
Does this comment require any action or it's okay to resolve it?

Reply:
I would prefer that we didn't say it was anything with more than 1 thread. It is one possible use for it but probably not the main one.

Reply:
Changed to `used to run tests that are expected to run across multiple ticks.`